From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: edac_core: Print debug messages at release calls
References: fate#311968
Patch-Mainline: yes
Git-commit: bbc560ae677c0f4d7ff8404a21409c99f35b297b


Signed-off-by: Thomas Renninger <trenn@suse.de>

This is important to track a nasty bug at the free logic.

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/edac_mc.c       |    2 ++
 drivers/edac/edac_mc_sysfs.c |    2 ++
 drivers/edac/i7core_edac.c   |    1 +
 3 files changed, 5 insertions(+)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/edac_mc.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/edac_mc.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/edac_mc.c
@@ -235,6 +235,8 @@ EXPORT_SYMBOL_GPL(edac_mc_alloc);
  */
 void edac_mc_free(struct mem_ctl_info *mci)
 {
+	debugf1("%s()\n", __func__);
+
 	edac_mc_unregister_sysfs_main_kobj(mci);
 }
 EXPORT_SYMBOL_GPL(edac_mc_free);
Index: linux-2.6.32-SLE11-SP1/drivers/edac/edac_mc_sysfs.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/edac_mc_sysfs.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/edac_mc_sysfs.c
@@ -713,6 +713,8 @@ fail_out:
  */
 void edac_mc_unregister_sysfs_main_kobj(struct mem_ctl_info *mci)
 {
+	debugf1("%s()\n", __func__);
+
 	/* delete the kobj from the mc_kset */
 	kobject_put(&mci->edac_mci_kobj);
 }
Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -2087,6 +2087,7 @@ static void __devexit i7core_remove(stru
 			edac_mc_del_mc(&i7core_dev->pdev[0]->dev);
 
 			/* Free data */
+			debugf1("%s: free structs\n");
 			kfree(mci->ctl_name);
 			edac_mc_free(mci);
 
